Missing PTR for `[N]*T` and `[N]*const T`

### Zig Version

0.12.0-dev.2763+7204eccf5

### Steps to Reproduce and Observed Behavior

Test code:
```zig
fn testPtr(comptime A: type, comptime B: type) void {
	_ = @TypeOf(@as(A, undefined), @as(B, undefined));
}
comptime {
	testPtr(*void, *const void); //works
	testPtr(@Vector(1, *void), @Vector(1, *const void)); //also works apparently
	testPtr([1]*void, [1]*const void); //compile error: incompatible types
}
```
`zig test main.zig` output:
```
main.zig:2:6: error: incompatible types: '[1]*void' and '[1]*const void'
 _ = @TypeOf(@as(A, undefined), @as(B, undefined));
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
main.zig:2:14: note: type '[1]*void' here
 _ = @TypeOf(@as(A, undefined), @as(B, undefined));
             ^~~~~~~~~~~~~~~~~
main.zig:2:33: note: type '[1]*const void' here
 _ = @TypeOf(@as(A, undefined), @as(B, undefined));
                                ^~~~~~~~~~~~~~~~~
main.zig:7:9: note: called from here
 testPtr([1]*void, [1]*const void); //compile error: incompatible types
 ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Expected Behavior

Peer-type resolution of the types should result in the array of pointer-to-const elements,
in line with vector types and the element types themselves.

--- a/src/Sema.zig
+++ b/src/Sema.zig
@@ -34231,7 +34231,21 @@ fn resolvePeerTypesInner(
                     .peer_idx_b = i,
                 } };
 
-                if (!ty.childType(mod).eql(elem_ty, mod)) {
+                const peer_elem_ty = ty.childType(mod);
+                if (!peer_elem_ty.eql(elem_ty, mod)) coerce: {
+                    const peer_elem_coerces_to_elem =
+                        try sema.coerceInMemoryAllowed(block, elem_ty, peer_elem_ty, false, mod.getTarget(), src, src);
+                    if (peer_elem_coerces_to_elem == .ok) {
+                        break :coerce;
+                    }
+
+                    const elem_coerces_to_peer_elem =
+                        try sema.coerceInMemoryAllowed(block, peer_elem_ty, elem_ty, false, mod.getTarget(), src, src);
+                    if (elem_coerces_to_peer_elem == .ok) {
+                        elem_ty = peer_elem_ty;
+                        break :coerce;
+                    }
+
                     return .{ .conflict = .{
                         .peer_idx_a = first_arr_idx,
                         .peer_idx_b = i,

--- a/test/behavior/cast.zig
+++ b/test/behavior/cast.zig
@@ -2201,6 +2201,23 @@ test "peer type resolution: pointer attributes are combined correctly" {
     try expectEqualSlices(u8, std.mem.span(@volatileCast(r3)), "baz");
 }
 
+test "peer type resolution: arrays of compatible types" {
+    if (builtin.zig_backend == .stage2_aarch64) return error.SkipZigTest; // TODO
+    if (builtin.zig_backend == .stage2_arm) return error.SkipZigTest; // TODO
+    if (builtin.zig_backend == .stage2_sparc64) return error.SkipZigTest; // TODO
+
+    var e0: u8 = 3;
+    var e1: u8 = 2;
+    var e2: u8 = 1;
+    const a = [3]*u8{ &e0, &e1, &e2 };
+    const b = [3]*const u8{ &e0, &e1, &e2 };
+
+    comptime assert(@TypeOf(a, b) == [3]*const u8);
+    comptime assert(@TypeOf(b, a) == [3]*const u8);
+
+    try expectEqual(@as(@TypeOf(a, b), a), b);
+}
+
 test "cast builtins can wrap result in optional" {
     if (builtin.zig_backend == .stage2_aarch64) return error.SkipZigTest; // TODO
     if (builtin.zig_backend == .stage2_arm) return error.SkipZigTest; // TODO

